### PR TITLE
Update HTTP calls retry logic

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**

### CODE REVIEW

File: src/oslogin_utils.cc [hunk 1]
```diff
@@ -428,7 +440,7 @@ bool HttpDo(const string& url, const string& data, string* response, long* http_
         return false;
       }
       curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, http_code);
-    } while (retry_count++ < kMaxRetries && *http_code == 500);
+    } while (retry_count++ < kMaxRetries && ShouldRetry(*http_code));
```
Comment (dorileo) [inline src/oslogin_utils.cc:443]: Are we idle retrying? Should we add a sleep and a back off strategy?

Comment (ChaitanyaKulkarni28) [reply src/oslogin_utils.cc:450]: Done

**Pull request merged**